src/lib/types/permissions.ts

```diff
@@ -141,4 +144,17 @@ export const ROOT_PERMISSION_CATEGORIES = [
             RELEASE_PLAN_TEMPLATE_UPDATE,
         ],
     },
+    {
+        label: 'Instance maintenance',
+        permissions: [UPDATE_MAINTENANCE_MODE, UPDATE_INSTANCE_BANNERS],
+    },
+];
+
+// Used on Frontend, to allow admin panel use for users with custom root roles
+export const MAINTENANCE_MODE_PERMISSIONS = [
+    ADMIN,
+    READ_ROLE,
+    READ_CLIENT_API_TOKEN,
+    READ_FRONTEND_API_TOKEN,
+    UPDATE_MAINTENANCE_MODE,
```

Comment:
How does this work? Do you need one of them or all of them? Seems a little strange to gain access to the full admin panel if you have read client API token only? Maybe I'm misunderstanding how this works?

Reply:
We don't have a way to selectively show some tabs (yet). It's a way to at least show all options, and `PermissionGuard` wrapper will do block further UI

![image](https://github.com/user-attachments/assets/bf9add88-9d92-4baa-9b7d-a762054d4bdb)